src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -675,19 +691,22 @@ void PolicyHandler::OnGetUserFriendlyMessage(
     const std::vector<std::string>& message_codes,
     const std::string& language,
     uint32_t correlation_id) {
-  LOG4CXX_AUTO_TRACE(logger_);
   POLICY_LIB_CHECK_VOID();
 
 #ifdef EXTERNAL_PROPRIETARY_MODE
   const std::string active_hmi_language =
       application_manager::MessageHelper::CommonLanguageToString(
           application_manager_.hmi_capabilities().active_ui_language());
+  policy_manager_lock_.Acquire();
   const std::vector<UserFriendlyMessage> result =
       policy_manager_->GetUserFriendlyMessages(
           message_codes, language, active_hmi_language);
+  policy_manager_lock_.Release();
 #else
+  policy_manager_lock_.Acquire();
   const std::vector<UserFriendlyMessage> result =
       policy_manager_->GetUserFriendlyMessages(message_codes, language);
+  policy_manager_lock_.Release();
```

Comment:
@SNiukalov Why do you make Lock and Release twice?

Reply:
@AByzhynar there is ifdef -> else -> endif sections.
But as for me you can use next way:
```
std::vector<UserFriendlyMessage> result;
#ifdef EXTERNAL_PROPRIETARY_MODE
  const std::string active_hmi_language =
      application_manager::MessageHelper::CommonLanguageToString(
          application_manager_.hmi_capabilities().active_ui_language());
  {
      sync_primitives::AutoLock lock(policy_manager_lock_);
      result = policy_manager_->GetUserFriendlyMessages(message_codes, language, active_hmi_language);
  }
#else
  {
      sync_primitives::AutoLock lock(policy_manager_lock_);
      result = policy_manager_->GetUserFriendlyMessages(message_codes, language);
  }
#endif // EXTERNAL_PROPRIETARY_MODE
```

Reply:
This code block is in the definition for EXTERNAL_PROPRIETARY_MODE.
In the code for EXTERNAL_PROPRIETARY_MODE,
before calling the PolicyManager method,
The methods of the ApplicationManager and HMICapabilities classes are called.
We don't use this object synchronize for them.

Reply:
@AByzhynar fix in: https://github.com/smartdevicelink/sdl_core/pull/2614/commits/793f178f719ad695595d1c8ad0ae40bcf265c9b5